### PR TITLE
Settings menu

### PR DESCRIPTION
**Pull request by 1000TurquoisePogs**
Initial settings menu which allows for customization of builtin monaco features such as the theme, font, and text behavior.
This creates a new menu item "edit" which has the submenu "preferences".
This opens up a tab which is for editing preferences of monaco, and you will see various properties which you can control by input boxes and dropdowns.
There's 3 buttons on this menu: 
* save which saves to the config service (and applies to the editor right away, and the editor uses these settings on startup)
* preview, which takes whatever json is in the text preview area and tries to apply it, if its valid
* reset, which gets the data from the config service again, in case you didnt like your changes.

### CODE REVIEW

**Comment by 1000TurquoisePogs** [comment]
I have the options "undo" and "redo" in the edit menu but they dont do anything yet. I'll either implement them or remove them from the PR, depending on if I have the time.

File: webClient/src/app/editor/code-editor/monaco-settings/monaco-settings.component.ts [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+  ngOnInit() {
+    this.editor = monaco.editor.create(this.monacoPreviewRef.nativeElement, this.config);
+    this.viewportEvents.resized.subscribe(()=> {
+      this.editor.layout()
+    });
+    monaco.editor.remeasureFonts();
+    this.setConfigFromConfigService();
+    setTimeout(()=> {
+      this.updateEditor();
+      this.editor.setModelLanguage('json');
```
Comment (lchudinov) [inline webClient/src/app/editor/code-editor/monaco-settings/monaco-settings.component.ts:199]: I get an error in the JS console: `setModelLanguage` is not a function. 
```typescript
this.editor.updateOptions({language: 'json'});
```
should work better.

**Review by lchudinov — CHANGES_REQUESTED** [review]
Good work. Aside from the errors in JS console I noticed that it doesn't update Preview when I press Reset.

**Comment by lchudinov** [comment]
When I change `LineNumbers` to `off` then Preview updates correctly. If I press `Revert` then nothing happens. If I press `Revert` second time Preview updates correctly, but the UI on the left still shows  `LineNumbers` is `off`.
revert

**Review by lchudinov — CHANGES_REQUESTED** [review]

File: webClient/src/app/editor/code-editor/monaco/monaco.component.ts [hunk 2]
```diff
@@ -28,7 +28,30 @@ const ReconnectingWebSocket = require('reconnecting-websocket');
...
-  @Input() options;
+  private _options: any;
+  
+  @Input()
+  get options(): any { return this._options; }
+  set options(options: any) {
+    if (!options) {return;}
+    this._options = options;
+    if (this.editor) {
+      console.log(this.editor);
```
Comment (lchudinov) [inline webClient/src/app/editor/code-editor/monaco/monaco.component.ts:39]: console.log

**Review by lchudinov — APPROVED** [review]
Excellent work👍.  I found a couple minor issue. See my PR if they are relevant: https://github.com/1000TurquoisePogs/zlux-editor/pull/1

**Pull request merged**